HelpSource/Classes/SequenceableCollection.schelp

```diff
@@ -889,6 +889,138 @@ copymethod:: SimpleNumber -atanh
 copymethod:: SimpleNumber -owensT
 
 
+subsection:: Testing
+
+method:: ==
+
+Returns true if each element is equal with its corresponding element in the receiver. See link::Classes/Object#Equality and Identity::.
+
+Discussion::
+Evaluate this code first to compare equality test and identity test:
+code::
+~array_a = [1, 2]; ~array_b = [1, 2]
+::
+
+list::
+## Equality:
+code::
+~array_a == ~array_b // checking for equality
+::
+teletype::
+-> true
+::
+It first checks if the two objects are identical, then they're not in the same class, then their sizes aren't the same, then there's an unequal element. If they are equal, then their emphasis::hash table indices:: are equal and identical:
+
+code::
+~array_a.hash
+~array_b.hash
+::
+code::
+~array_a.hash ==  ~array_b.hash
+::
+teletype::
+-> true
+::
+code::
+~array_a.hash === ~array_b.hash
+::
+teletype::
+-> true
+::
+
+## Identity
+
+code::
+~array_a === ~array_b // checking for identity
+::
+
+teletype::
+-> false
+::
+They are two different objects allocated at different addresses in virtual memory:
+
+list::
+## code::~array_a::
+
+code::
+~array_a.dump
+::
+
+returns:
+teletype::
+Instance of Array {    (0x1256a70b8, gc=38, fmt=01, flg=00, set=02)
+  indexed slots [2]
+      0 : Integer 1
+      1 : Integer 2
+}
+-> [1, 2]
+::
+
+The first two parts of the parenthesis pair above, i.e. teletype::0x1256a70b8:: and teletype::gc=38::, may be different when evaluating the code code::a.dump::, but their values are determined when evaluating the code code::~array_b = [1, 2]::. (see: link::#-dump::)
+
+##  code::~array_b::
+code::
+~array_b.dump
+::
+
+returns:
+teletype::
+Instance of Array {    (0x152d02da8, gc=70, fmt=01, flg=00, set=02)
+  indexed slots [2]
+      0 : Integer 1
+      1 : Integer 2
+}
+-> [1, 2]
+::
+
+The first two parts of the parenthesis pair above, i.e. teletype::0x152d02da8:: and teletype::gc=70::, may be different when evaluating the code code::b.dump::, but their values are determined when evaluating the code code::~array_b = [1, 2]::. (see: link::#-dump::)
+::
+
+The hexadecimal digits addresses indicate the address on virtual memory.
+
+list::
+## The address of the object associated with the variable code::~array_a:::
+
+teletype::
+0x1256a70b8
+::
+
+## The address of the object associated with the variable code::~array_b:::
+
+teletype::
+0x152d02da8
+::
+::
+
```

Comment:
I think this whole section about memory is unnecessary here. There could be a dedicated section about memory in supercollider, but it should go in SequeneableCollection.

Reply:
No, this should be removed. Memory is an implementation detail and should be avoided.

Reply:
OK. I see.

@telephon
What do you think? If I create a new file, should this part be moved there or removed as @JordanHendersonMusic suggests?

Reply:
I don't think it would be possible to talk about memory without discussing slots as you'd need to account for numbers which live in slots and aren't allocated.